packages/file_selector/file_selector_platform_interface/lib/src/types/x_file/html.dart

```diff
@@ -0,0 +1,161 @@
+        _lastModified = lastModified,
+        super(path);
+
+  /// Construct an XFile from its data
+  XFile.fromData(
+      Uint8List bytes, {
+        this.mimeType,
+        this.name,
+        int length,
+        DateTime lastModified,
+        @visibleForTesting XFileTestOverrides overrides,
+      })  : _data = bytes,
+        _length = length,
+        _overrides = overrides,
+        _lastModified = lastModified,
+        super('') {
+    Blob blob;
+    if (mimeType == null) {
+      blob = Blob([bytes]);
+    } else {
+      blob = Blob([bytes], mimeType);
+    }
+    this.path = Url.createObjectUrl(blob);
+  }
+
+
+
+  Future<Uint8List> get _bytes async {
+    if (_data != null) {
+      return Future.value(UnmodifiableUint8ListView(_data));
+    }
+    return http.readBytes(path);
+  }
+
+  @override
+  Future<int> length() async {
+    return _length ?? (await _bytes).length;
+  }
+
+  @override
+  Future<String> readAsString({Encoding encoding = utf8}) async {
+    return encoding.decode(await _bytes);
+  }
+
+  @override
+  Future<Uint8List> readAsBytes() async {
+    return Future.value(await _bytes);
+  }
+
+  @override
+  Stream<Uint8List> openRead([int start, int end]) async* {
+    final bytes = await _bytes;
+    yield bytes.sublist(start ?? 0, end ?? bytes.length);
+  }
+
+  /// Saves the data of this XFile at the location indicated by path.
+  /// For the web implementation, the path variable is ignored.
+  void saveTo(String path) async {
+    // Create a DOM container where we can host the anchor.
+    _target = _ensureInitialized('__x_file_dom_element');
+
+    // Create an <a> tag with the appropriate download attributes and click it
+    final AnchorElement element = createAnchorElement(this.path, this.name);
+
+    _addElementToDomAndClick(element);
+  }
+
+  /// Create anchor element with download attribute
+  @visibleForTesting
+  AnchorElement createAnchorElement(String href, String suggestedName) {
+    if (_hasTestOverrides && _overrides.createAnchorElement != null) {
+      return _overrides.createAnchorElement(href, suggestedName);
+    }
+
+    final element = AnchorElement(href: href);
+
+    if (suggestedName == null) {
+      element.download = 'download';
+    } else {
+      element.download = suggestedName;
+    }
+
+    return element;
+  }
+
+  void _addElementToDomAndClick(Element element) {
+    // Add the file input element and click it
+    // All previous elements will be removed before adding the new one
+    _target.children.clear();
+    _target.children.add(element);
+    element.click();
+  }
+
+  /// Initializes a DOM container where we can host elements.
+  Element _ensureInitialized(String id) {
+    var target = querySelector('#${id}');
+    if (target == null) {
+      final Element targetElement =
+      Element.tag('flt-x-file')..id = id;
+
+      querySelector('body').children.add(targetElement);
+      target = targetElement;
+    }
+    return target;
+  }
```

Comment:
Can this bunch of methods live outside of the XFile itself? They look like bare helpers, since they don't need "this" for anything :)